cadl-extension/src/main/java/com/azure/cadl/Main.java

```diff
@@ -0,0 +1,140 @@
+import com.google.googlejavaformat.java.Formatter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+import org.yaml.snakeyaml.representer.Representer;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+public class Main {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Main.class);
+
+    // java -jar target/azure-cadl-extension-jar-with-dependencies.jar
+
+    public static void main(String[] args) throws IOException {
+        // parameters
+        String inputYamlFileName = "cadl-tests/cadl-output/code-model.yaml";
+        String outputFolder = "cadl-tests/cadl-output/java/";
+        String namespace = "com.azure.cadl";
+        if (args.length >= 1) {
+            inputYamlFileName = args[0];
+        }
+        if (args.length >= 2) {
+            outputFolder = args[1];
+            if (!outputFolder.endsWith("/")) {
+                outputFolder += "/";
+            }
+        }
+
+        LOGGER.info("Code model file: {}", inputYamlFileName);
+        LOGGER.info("Output folder: {}", outputFolder);
+
+        // load code-model.yaml
+        CodeModel codeModel = loadCodeModel(inputYamlFileName);
+
+        if (codeModel.getLanguage().getJava() != null && !CoreUtils.isNullOrEmpty(codeModel.getLanguage().getJava().getNamespace())) {
+            namespace = codeModel.getLanguage().getJava().getNamespace();
+        }
+        LOGGER.info("Namespace: {}", namespace);
+
+        // initialize plugin
+        CadlPlugin cadlPlugin = new CadlPlugin(namespace);
+
+        // transform cod emodel
+        codeModel = new Transformer().transform(codeModel);
+
+        // map to client model
+        Client client = Mappers.getClientMapper().map(codeModel);
+
+        // template
+        JavaPackage javaPackage = cadlPlugin.writeToTemplates(codeModel, client, JavaSettings.getInstance());
+
+        LOGGER.info("Count of Java files: {}", javaPackage.getJavaFiles().size());
+        LOGGER.info("Count of XML files: {}", javaPackage.getXmlFiles().size());
+        LOGGER.info("Count of text files: {}", javaPackage.getTextFiles().size());
+
+        // write output
+        Formatter formatter = new Formatter();
+        for (JavaFile javaFile : javaPackage.getJavaFiles()) {
+            String content = javaFile.getContents().toString();
+            try {
+                content = formatter.formatSourceAndFixImports(content);
+            } catch (Exception e) {
+                LOGGER.error("Failed to format file: {}", outputFolder + javaFile.getFilePath(), e);
+//                continue;
+            }
+            cadlPlugin.writeFile(outputFolder + javaFile.getFilePath(), content, null);
+        }
```

Comment:
I'd recommend copying the logic used in Javagen or Postprocessor that speeds up formatting and writing the generated files

Reply:
Agree. Got it in your PR but copied this from old Javagen.